File: 프로그래머스/lv1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        List<Integer> answerList = new ArrayList<>();
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] % divisor == 0) {
                answerList.add(arr[i]);
            } 
        }
        if (answerList.isEmpty()) {
            return new int[]{-1};
        }

        // List를 배열로 변환
        int[] answer = new int[answerList.size()];
        for (int i = 0; i < answerList.size(); i++) {
            answer[i] = answerList.get(i);
        }

        // 배열 정렬
        Arrays.sort(answer);
        return answer;
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.py
def solution(arr, divisor):
    answer = []
    for i in range(len(arr)):  # 0부터 시작하는 루프
        if arr[i] % divisor == 0:
            answer.append(arr[i])  # i가 아닌 arr[i]를 추가 (값을 추가)
    
    if not answer:  # answer 리스트가 비어있으면
        return [-1]  # -1을 반환
    
    return sorted(answer)
